src/i965_encoder.c

```diff
@@ -1453,6 +1468,16 @@ intel_enc_hw_context_init(VADriverContextP ctx,
             encoder_context->quality_range = ENCODER_QUALITY_RANGE_AVC;
         } else
             encoder_context->quality_range = ENCODER_QUALITY_RANGE;
+
+        if (obj_config->entrypoint == VAEntrypointFEI) {
+            encoder_context->fei_enabled = 1;
```

Comment:
I mean maybe we need to change it to:
if (obj_config->entrypoint == VAEntrypointFEI && HAS_FEI_H264_ENCODING(i965)) 
    encoder_context->fei_enabled = 1;
Then No need to check IS_SKL() in gen9_avc_encode.c, just like such code:
+        if (!encoder_context->fei_enabled) {
 +            generic_ctx->enc_kernel_ptr = (void *)skl_avc_encoder_kernels;
 +            generic_ctx->enc_kernel_size = sizeof(skl_avc_encoder_kernels);
 +        } else if (IS_SKL(i965->intel.device_info)) {
 +            generic_ctx->enc_kernel_ptr = (void *)skl_avc_fei_encoder_kernels;
 +            generic_ctx->enc_kernel_size = sizeof(skl_avc_fei_encoder_kernels);
 +        }
How do you think?